### PR TITLE
Fixes #2877: removed mockito-inline as subproject and actual subproject

### LINKED ISSUE #2877: Mockito-inline dependency after upgrade 5.0.0
After reading that mockito switched to mockito-inline by default (with 5.0.0) i expected either mockito-core to have a mockito-inline dependency, or mockito-inline has been merged into mockito-core.

Interestingly neither of this is the case (since mockito-inline has a 5.0.0 release and no mockito-inline transitive dependency is found in https://mvnrepository.com/artifact/org.mockito/mockito-core/5.0.0 )

The concrete question is, how should we go about the current explicit dependency on mockito-inline. Should it be removed with 5.0.0 or should it stay?

### PR DESCRIPTION
**Pull request by krzyk**
Fixes #2877 as suggested in https://github.com/mockito/mockito/issues/2877 removed inline from settings.gradle.kts and also the inline subproject itself.
Also removed groovyInlineTest which was testing this subproject.

## Checklist

 - [x] Read the [contributing guide](https://github.com/mockito/mockito/blob/main/.github/CONTRIBUTING.md)
 - [x] PR should be motivated, i.e. what does it fix, why, and if relevant how
 - [x] If possible / relevant include an example in the description, that could help all readers
       including project members to get a better picture of the change
 - [x] Avoid other runtime dependencies
 - [x] Meaningful commit history ; intention is important please rebase your commit history so that each
       commit is meaningful and help the people that will explore a change in 2 years
 - [x] The pull request follows coding style
 - [x] Mention `Fixes #<issue number>` in the description _if relevant_
 - [x] At least one commit should mention `Fixes #<issue number>` _if relevant_

### CODE REVIEW

**Comment by krzyk** [comment]
Sorry I was a bit surprised that ANDROID_HOME needs to be set to run part of android tests.

**Comment by TimvdLippe** [comment]
Seems like this would reduce our test coverage by quite a lot. Can we keep the inline test suite and rather it depending on `mockito-inline`, it depends on `mockito-core` instead which already uses the inline mockmaker by default.

**Comment by krzyk** [comment]
@TimvdLippe Sure, I replaced the dependency in groovyInline tests

**Comment by krzyk** [comment]
@TimvdLippe Yes, I misunderstood your previous comment. Now all the tests are back.

File: settings.gradle.kts [hunk 1]
```diff
@@ -3,11 +3,9 @@ plugins {
 }
 
 include("subclass",
-    "inline",
```
Comment (TimvdLippe) [inline settings.gradle.kts:6]: While we need to delete the source files of this artifact, we do need to keep this module so it runs the tests. Please run the code coverage locally to see what it is doing, as it is reporting missing usage.

I recommend renaming this module to `inlineTest`. Then, we need to make sure we don't accidentally publish it, but iirc by appending Test to it, it works. I can double check that later.

Comment (krzyk) [reply settings.gradle.kts:6]: Oops, forgot about that. I checked the coverage locally and the html report doesnt' show decimals so I assumed 85% was it.
Now after I readded the renamed project it is at 86%.

File: subprojects/inlineTest/inlineTest.gradle
Comment (TimvdLippe) [inline subprojects/inlineTest/inlineTest.gradle:1]: We need to remove the `java-library.gradle` line from this file, to make sure we don't accidentally start publishing this artifact. I think this should can be the replacement:

```
apply(from = "$rootDir/gradle/dependencies.gradle")

plugins {
    java
}
```

See https://github.com/mockito/mockito/blob/19965ff915bf6c19eeb350f07d8ddfefdb95b1a0/subprojects/memory-test/memory-test.gradle as an example

Comment (krzyk) [reply subprojects/inlineTest/inlineTest.gradle:1]: OK, applied

**Review by TimvdLippe — APPROVED** [review]
Yup, that looks good to me! Thanks for the PR and the updates 😄

**Pull request merged**